src/Schema/Index.php

```diff
@@ -101,10 +109,14 @@ public function getQuotedColumns(AbstractPlatform $platform): array
         foreach ($this->_columns as $column) {
             $length = array_shift($subParts);
 
-            $quotedColumn = $column->getQuotedName($platform);
+            if ($this->isFunctional()) {
+                $quotedColumn = $column->getName();
```

Comment:
A functional column does not have a name. We should find a better representation for such columns.